src/workers/newNotificationV2Mail.ts

```diff
@@ -113,12 +113,12 @@ const notificationToTemplateData: Record<NotificationType, TemplateDataFunc> = {
       .findOneBy({ id: notification.referenceId })) as ArticlePost;
     const submission = await con
       .getRepository(Submission)
-      .findOneBy({ url: post.url, userId: user.id });
+      .findOneByOrFail({ url: post.url!, userId: user.id });
```

Comment:
Do we want hard fails here?
(not sure how CDC will handle it vs if it's manually early return?)

Reply:
Below it uses `submission.createdAt` so it would fail either way if submission was null, I was thinking to avoid `findOneByOrFail` and use `!` (force unwrap) but think this is better.

From what I can see it will just log `failed to send notification email` as usual, so it should not go into spiral of retrying.

Reply:
Alright, let's keep like this, just keep an eye out on merging :)

Reply:
I will 🫡